### ISSUE TITLE
Add desc_addname to no smart-quotes list

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#3826]
In detail, please refer https://github.com/sphinx-doc/sphinx/pull/3816#discussion_r118893735

Sphinx provides following TextElement subclasses.
To make smart-quotes completely resolved, we have to check how these classes are used and should or should not be smart-quoted.
```
$ grep TextElement sphinx/addnodes.py
class desc_signature(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_signature_line(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_addname(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_type(nodes.Part, nodes.Inline, nodes.TextElement):
        return ' -> ' + nodes.TextElement.astext(self)
class desc_name(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_parameterlist(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_parameter(nodes.Part, nodes.Inline, nodes.TextElement):
class desc_optional(nodes.Part, nodes.Inline, nodes.TextElement):
        return '[' + nodes.TextElement.astext(self) + ']'
class desc_annotation(nodes.Part, nodes.Inline, nodes.TextElement):
class versionmodified(nodes.Admonition, nodes.TextElement):
class production(nodes.Part, nodes.Inline, nodes.TextElement):
class index(nodes.Invisible, nodes.Inline, nodes.TextElement):
class centered(nodes.Part, nodes.TextElement):
class abbreviation(nodes.Inline, nodes.TextElement):
class manpage(nodes.Inline, nodes.TextElement):
```

**Comment by jfbu** [comment]
(edited)

Surely ``desc_annotation`` which was overlooked at #3816.

Indeed, with this `foocode.py` module

```
#!/usr/bin/env python
# coding: utf-8
"""
Created on 30 May 2017
@author: Foo Bar
"""

teststring = '' #: this serves only to test if smart quotes are applied

```

and 

```
.. automodule:: foocode
   :members:
```

one sees smart quotes are applied (``<em class="property"> = ‘’</em>``) below:

```html
<dl class="data">
<dt id="foocode.teststring">
<code class="descclassname">foocode.</code><code class="descname">teststring</code><em class="property"> = ‘’</em><a class="headerlink" href="#foocode.teststring" title="Permalink to this definition">¶</a></dt>
<dd><p>this serves only to test if smart quotes are applied</p>
</dd></dl>
```

This is with stable HEAD at 2b26520 which incorporates the a3bdf1d0 and
0b0bbfcb Docutils backports. Earlier, there would have been a pair of  closing curly double quotes ``”`` in place of ``‘’``.

Of course,  we want ``''`` straight quotes in output.

**Comment by jfbu** [comment]
At f99fe20e I have updated list of text parent nodes to skip. I am not sure if `desc_signature_line` can be a direct such parent, but I played it safe. 

Note: smart quotes are applied inside docstrings and comments of documented classes or functions. Tested with:

```
class Foo:
    """Docstring for 'my' class Foo."""

    #: Doc "comment" for class attribute Foo.bar.
    #: It can have multiple lines.
    bar = "1"

    flox = "1.5"   #: Doc "comment" for Foo.flox. One line only.

    baz = "2"
    """Docstring for class attribute Foo.baz."""

    def __init__(self):
        #: Doc comment for "instance" attribute qux.
        self.qux = "3"

        self.spam = "4"
        """Docstring for 'instance' attribute spam."""
```

All quotes from inside the docstrings and comments are smartified. But not (now) those from Python syntax...

**Comment by tk0miya** [comment]
`:manpage:` also should be marked as "not smarty quotes".
I will do that after #3875.

**Comment by tk0miya** [comment]
Fixed at 1f8604e48.
